blockchain/processing.py

```diff
@@ -340,6 +306,34 @@ def _execute_phase_1(self, config, current_block_id):
                 tx["header"]["status"] = "rejected"
                 transaction_db.update_transaction(tx)
 
+    def handle_transaction(self, txn):
+        """
+            check if the given transaction falls under one of these smart contract types and execute
+            the appropriate smart contract provisioning function on it
+         """
+        txn_type = txn["header"]["transaction_type"]
+        # reserved transaction smart contracts
+        rtsc = self.scp.rtsc
+        if txn_type in rtsc:
+            if rtsc[txn["header"]["transaction_type"]](txn):
+                return True
+        elif txn_type == "TT_PROVISION_TSC":
```

Comment:
I'm thinking that we'll want the caller to specify the SC implementation, with TT naming being same for all (per SC type).